src/System.Private.ServiceModel/tests/Common/Infrastructure/src/ConditionalTestDetectors.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Infrastructure.Common
+{
+    // This class contains helper "detector" methods to detect specific
+    // conditions about the environment in which the tests are run.
+    // They are used for [ConditionalFact] and [ConditionalTheory]
+    // tests.  They are called at most once, and the return value cached.
+    internal static class ConditionalTestDetectors
+    {
+        private const string ClientCertificateSubject = "WCF Client Certificate";
+        private const string CertificateIssuer = "DO_NOT_TRUST_WcfBridgeRootCA";
+
+        public static bool IsRootCertificateInstalled()
+        {
+            return IsCertificateInStore(StoreName.Root, StoreLocation.LocalMachine, CertificateIssuer) ||
+                   IsCertificateInStore(StoreName.Root, StoreLocation.CurrentUser, CertificateIssuer);
+        }
+
+        public static bool IsClientCertificateInstalled()
+        {
+            return IsCertificateInStore(StoreName.My, StoreLocation.LocalMachine, CertificateIssuer) ||
+                   IsCertificateInStore(StoreName.My, StoreLocation.CurrentUser, CertificateIssuer);
+        }
```

Comment:
Need to use client certificate subject to find out whether client cert is installed.